### PR TITLE
cmd/openshift-install/upi: Add a user-provided-infrastructure subcommand

### PR DESCRIPTION
**Pull request by wking**
To support:

1. Call `create FIXME` to create the bootstrap Ignition config and admin kubeconfig, #1119.
2. Creates all required cluster resources, after which the cluster will being bootstrapping.
3. Call `user-provided-infrastructure bootstrap-complete` to wait until the bootstrap phase has completed.
4. Destroy the bootstrap resources.
5. Call `user-provided-infrastructure finish` to wait until the cluster finishes deploying its initial version.  This also retrieves the router certificate authority from the cluster and inserts it into the admin kubeconfig.

CC @abhinavdahiya, @staebler, @cuppett

### CODE REVIEW

**Comment by wking** [comment]
> 1. Call `create FIXME` to create the bootstrap Ignition config and admin kubeconfig, #1119.

I'd forgotten that `create ignition-configs` creates the admin kubeconfig since #1008, so this doesn't need to be a FIXME.  #1119 just has the discussion about renaming the `ignition-configs` target to something more appropriate now that it's going to become just "the assets you need for UPI".

**Comment by cuppett** [comment]
I was able to use this and detect the bootstrap complete event. Here are a couple notes during my session:

[scuppett@cuppett installer]$ bin/openshift-install user-provided-infrastructure bootstrap-complete
INFO Waiting up to 30m0s for the Kubernetes API...

Notes: May be useful to identify the target address here. Just in case there's a weird mismatch between what the installer is thinking and the reality of the currently configured UPI.

INFO API v1.12.4+bf1e6bd up                       
INFO Waiting up to 30m0s for the bootstrap-complete event... 

Notes: Should we tell them to destroy the bootstrap here?

[scuppett@cuppett installer]$ bin/openshift-install user-provided-infrastructure finish
INFO Waiting up to 30m0s for the cluster to initialize...

**Comment by wking** [comment]
> May be useful to identify the target address here.
> ...
> Should we tell them to destroy the bootstrap here?

Both done with 5b3626e09 -> 45fbb29a2.

File: docs/user/customization.md [hunk 1]
```diff
@@ -184,7 +184,7 @@ For example:
     02-master-kubelet                                                                        2.2.0             137m
     01-worker-container-runtime                                 3.11.0-744-g5b05d9d3-dirty   2.2.0             137m
     01-worker-kubelet                                           3.11.0-744-g5b05d9d3-dirty   2.2.0             137m
-    99-master-3c81ffa3-3b8d-11e9-ac1e-52fdfc072182-registries   3.11.0-744-g5b05d9d3-dirty                     133m
+    99-master-3c81ffa3-3b8d-11e9-ac1e-52fdfc072182-registries   3.11.0-744-g5b05d9d3-dirty               docs/user/customization.md      133m
```
Comment (abhinavdahiya) [inline docs/user/customization.md]: un-related change.

Comment (wking) [reply docs/user/customization.md]: Oops, rouge paste.  Will fix shortly

Comment (wking) [reply docs/user/customization.md]: > un-related change.

Fixed with fefd4be7b -> bfc40f87f.

**Pull request merged**